### PR TITLE
Gradient clipping by norm has different semantics in tf.keras.optimizers against keras.optimizers.

### LINKED ISSUE #29108: Gradient clipping by norm has different semantics in tf.keras.optimizers against keras.optimizers
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 16.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: No
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 1.13.1
- Python version: 3.6
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: CUDA 10.0 + cuDNN 7.4
- GPU model and memory: NVIDIA Tesla V100 32GB

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with: 1. TF 1.0: `python -c "import
tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"` 2. TF 2.0: `python -c
"import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**

The gradient clipping mechanism is implemented in the abstract class, tf.keras.optimizers.Optimizer (keras.optimizers.Optimizer) so that every optimizer inherited from the class supports it.
We find different semantics of the implementations between tf.keras and keras.io.

In tf.keras, the gradient norm is calculated and clipped **per gradient tensor**.

https://github.com/tensorflow/tensorflow/blob/bb8cf258bc87f68612a5e032d5b4def0d3c52566/tensorflow/python/keras/optimizers.py#L98-L99

But in keras.io, the gradient norm is calculated globally **across all gradient tensors**.

```python
        if hasattr(self, 'clipnorm') and self.clipnorm > 0:
            norm = K.sqrt(sum([K.sum(K.square(g)) for g in grads]))
            grads = [clip_norm(g, self.clipnorm, norm) for g in grads]
```
Code link: https://github.com/keras-team/keras/blob/eab1b5bcdf105746ede02d2eb8a5cb3ca359b1b5/keras/optimizers.py#L96-L98

IMO, tf.keras should adopt the latter method since
(1) The paper introducing gradient clipping (https://arxiv.org/abs/1211.5063) suggests the latter one, and
(2) tf.keras should be compliant with keras.io.

### PR DESCRIPTION
**Pull request by chenchc**
Fixes #29108 .

### CODE REVIEW

**Comment by gbaned** [comment]
@chenchc Did you get a chance to look on reviewer comments? Please let us know on the update. Thanks!

**Comment by gbaned** [comment]
@chenchc Gentle ping to check the reviewer comments. Thanks!

**Comment by rthadur** [comment]
I'm going to go ahead and close this PR, because it seems to have stalled. If you're still interested in pursing this (and responding to my comments), please feel free to reopen!
